src/model/kubernetes.js

```diff
@@ -0,0 +1,329 @@
+const KubeClient = require('kubernetes-client').Client;
+const core = require('@actions/core');
+const base64 = require('base-64');
+
+class Kubernetes {
+  static async runBuildJob(buildParameters, baseImage) {
+    // uses default kubeconfig location/env variable
+    const kubeClient = new KubeClient();
+    await kubeClient.loadSpec();
+
+    const buildId = Kubernetes.uuidv4();
+    const pvcName = `unity-builder-pvc-${buildId}`;
+    const secretName = `build-credentials-${buildId}`;
+    const jobName = `unity-builder-job-${buildId}`;
+
+    Object.assign(this, {
+      kubeClient,
+      buildId,
+      buildParameters,
+      baseImage,
+      pvcName,
+      secretName,
+      jobName,
+    });
+
+    await Kubernetes.createSecret();
+    await Kubernetes.createPersistentVolumeClaim();
+    await Kubernetes.createBuildJob();
```

Comment:
As we talked about some "scheduler" and "license orchestrator" concepts before, I'd be in favor of calling this `scheduleBuildJob`

Reply:
Totally agree!